components/builder-graph/src/package_graph_trait.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::protocol::originsrv;
+
+#[derive(Debug)]
+pub struct Stats {
+    pub node_count:     usize,
+    pub edge_count:     usize,
+    pub connected_comp: usize,
+    pub is_cyclic:      bool,
+}
+
+pub trait PackageGraphTrait: Send + Sync {
+    fn build(&mut self,
```

Comment:
Maybe this could be `new`? If having empty graphs is useful, you could add an `empty` trait method which calls `build` on an empty slice.